src/python/pants/backend/javascript/subsystems/nodejs_tool_test.py

```diff
@@ -85,8 +91,8 @@ def test_execute_process_with_package_manager(
     "package_manager, version",
     [
         pytest.param("yarn", "1.22.22", id="yarn"),
-        pytest.param("npm", "10.8.2", id="npm"),
-        pytest.param("pnpm", "9.5.0", id="pnpm"),
+        pytest.param("npm", "10.9.0", id="npm"),
+        pytest.param("pnpm", "9.12.3", id="pnpm"),
```

Comment:
Just checking before merging: can you expand a bit on why these changed?

(I'd like to be sure that we don't have to update them again when new versions are released, so that these tests don't depend on external state.)

Reply:
Yeah, those are now picked up by the version installed via corepack. I'm not really sure on how we could lock them, when running the activate command for corepack perhaps?

Reply:
Hm, thanks.

It's definitely unexpected to me that setting `--nodejs-package-managers={...}` with explicit versions doesn't end up using those exact versions, I think we may need to fix this.

@tobni, any opinions/advice?

Reply:
ah, looking at  https://github.com/nodejs/corepack?tab=readme-ov-file#utility-commands suggests that we could run like, for instance, `args=("npm@10.8.2", ...)`. So, maybe:

1. add a method for getting the `...@...` installation specification to `PackageManager`, e.g.:
    ```python
    def spec(self) -> str:
        return f"{self.name}@{self.version}` if self.version is not None else self.name
    ```
2. invoke the process with `args=(pkg_manager.spec(), ...)`

Reply:
Ah, nice find. I'll try and see if I can get to it later this week

Reply:
Just quickly tested this and it seems to work for all package managers except yarn that ends up with this error ```
error Couldn't find a package.json file in "/private/tmp/pants-sandbox-pPJw4m"
```
Might be connected to https://github.com/nodejs/corepack/issues/334